xaitk_saliency/utils/tools/gen_detector_sal.py

```diff
@@ -0,0 +1,148 @@
+import numpy as np
+from typing import Optional, Collection, Iterable, Tuple, Dict, Hashable, Callable, Any
+import logging
+
+from smqtk_detection.interfaces.detect_image_objects import DetectImageObjects
+from smqtk_detection.utils.bbox import AxisAlignedBoundingBox
+
+from xaitk_saliency.interfaces.gen_detector_prop_sal import GenerateDetectorProposalSaliency
+from xaitk_saliency.interfaces.perturb_image import PerturbImage
+from xaitk_saliency.utils.masking import occlude_image_batch
+from xaitk_saliency.utils.detection import format_detection
+
+logging.basicConfig(level = logging.INFO)
```

Comment:
Avoid initializing logging in the module, especially at the raw module level. This is something that should occur on a per-tool basis as it applies to that tool. E.g. any adding of handlers to a logger (which is what this does) should be in the CLI logic.